target/linux/realtek/base-files/etc/init.d/hwmon_fancontrol

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh /etc/rc.common
+
+START=18
+
+boot() {
+	# configuring onboard temp/fan controller to run the fan on its own
+	# for more information, please read https://www.kernel.org/doc/Documentation/hwmon/sysfs-interface
+
+	case $(board_name) in
+	plasmacloud,psx28)
+		path_to_hwmon="$(find /sys/devices/platform/1b000000.switchcore/1b000000.switchcore:i2c@100c/i2c-4/4-002e/hwmon/ -type d -maxdepth 1 -mindepth 1 -name 'hwmon*')"
+
+		# adt7476 fan control chip
+		# 3 temp sensors. Set to lowest speed at 35C and max speed at 70C
+		# set critical temp to 80 as per the datasheet
+		echo 7 > "$path_to_hwmon/pwm2_auto_channels_temp"
```

Comment:
Make this coding a device specific subfunction to keep the boot() function clean.